packages/sign-client/src/controllers/engine.ts

```diff
@@ -1137,7 +1137,7 @@ export class Engine extends IEngine {
     const { id, topic, params, verifyContext } = pendingRequest;
     const expiry =
       params.request.expiryTimestamp || calcExpiry(ENGINE_RPC_OPTS.wc_sessionRequest.req.ttl);
-    this.client.core.expirer.set(id, expiry);
+    if (expiry) this.client.core.expirer.set(id, expiry);
```

Comment:
Why do we need to guard the value as truthy if it should always have a default?

Is this what is wanted?
```suggestion
    if (expiry > 0) this.client.core.expirer.set(id, expiry);
```

Reply:
the condition was copied along the `expirer.set` from other flows where expiry didn't have default value, removed it